SCons/Tool/ninja/__init__.py

```diff
@@ -214,7 +214,7 @@ def ninja_generate_deps(env):
 
     # This adds the required flags such that the generated compile
     # commands will create depfiles as appropriate in the Ninja file.
-    if env["PLATFORM"] == "win32":
+    if env["PLATFORM"] == "win32" and env.get('CC') != 'gcc':
```

Comment:
could you check:

`if 'mingw' in env['TOOLS']:`

Reply:
done in latest commit